spec/clients_spec.rb

```diff
@@ -130,4 +130,149 @@
+                                              app_secret: config['app_secret']) }
+    before(:each) do
+      @af = stub_request(:any, /#{AppFigures::API::BASE_URL}\/*/)
+      @date = '2016-01-01'
+    end
+
+    it 'get usage' do
+      body = {'usage'=>10}
+      @af.to_return(body: body.to_json,
+                    status: 200,
+                    headers: {'X-Request-Limit' => 1000, 'X-Request-Usage' => 10})
+      resp = client.usage
+      expect(resp).not_to be_nil
+      expect(resp).to include(body)
+    end
+
+    it 'get products' do
+      body = {'id'=> 1234567, 'name'=>'myapp'}
+      @af.to_return(body: body.to_json,
+                    status: 200,
+                    headers: {'X-Request-Limit' => 1000, 'X-Request-Usage' => 10})
+      resp = client.products
+      expect(resp).not_to be_nil
+      expect(resp).to include(body)
+    end
+
+    it 'get sales' do
+      body = {'id'=> 1234567, 'name'=>'myapp'}
+      @af.to_return(body: body.to_json,
+                    status: 200,
+                    headers: {'X-Request-Limit' => 1000, 'X-Request-Usage' => 10})
+      resp = client.sales
+      expect(resp).not_to be_nil
+      expect(resp).to include(body)
+    end
+
+    it 'get revenue' do
+      body = {'id'=> 1234567, 'name'=>'myapp'}
+      @af.to_return(body: body.to_json,
+                    status: 200,
+                    headers: {'X-Request-Limit' => 1000, 'X-Request-Usage' => 10})
+      resp = client.revenue
+      expect(resp).not_to be_nil
+      expect(resp).to include(body)
+    end
+
+    it 'get ads' do
+      body = {'id'=> 1234567, 'name'=>'myapp'}
+      @af.to_return(body: body.to_json,
+                    status: 200,
+                    headers: {'X-Request-Limit' => 1000, 'X-Request-Usage' => 10})
+      resp = client.ads
+      expect(resp).not_to be_nil
+      expect(resp).to include(body)
+    end
+
+    it 'get rank' do
+      body = {'rank'=> 15}
+      @af.to_return(body: body.to_json,
+                    status: 200,
+                    headers: {'X-Request-Limit' => 1000, 'X-Request-Usage' => 10})
+      expect{client.ranks}.to raise_error(ArgumentError)
+      resp = client.ranks(ids:'123;456')
+      expect(resp).not_to be_nil
+      expect(resp).to include(body)
+    end
+
+    it 'get featured invalid mode' do
+      expect{ client.featured(mode: 'invalid mode') }.to raise_error(ArgumentError)
+    end
+
+    it 'get featured summary' do
+      body = {'id'=> 1234567, 'paths'=> %w(apple free handheld)}
+      @af.to_return(body: body.to_json,
+                    status: 200,
+                    headers: {'X-Request-Limit' => 1000, 'X-Request-Usage' => 10})
+      [
+        [nil,   nil],
+        [nil,   @date],
+        [@date, nil],
+      ].each do |start, end_date|
+        expect{ client.featured(mode: 'summary', start_date: start, end_date: end_date) }.to raise_error(ArgumentError)
+      end
```

Comment:
Let's make test simpler.
Here we test failure case and success case within one test case.
Separating them into two test cases is better when running testing.